server/src/inmantals/server.py

```diff
@@ -290,6 +276,10 @@ async def initialize(
             if rootUri is None:
                 raise InvalidExtensionSetup("No workspace folder or rootUri specified.")
             workspaceFolders = [rootUri]
+            workspace_folder = lsp_types.WorkspaceFolder(uri=rootUri, name=os.path.dirname(urlparse(rootUri).path))
+        else:
+            workspace_folder = lsp_types.WorkspaceFolder(uri=workspaceFolders[0]["uri"], name=workspaceFolders[0]["name"])
```

Comment:
If the data is invalid this might fail with a `KeyError` so it's better to pass it as a whole to `pydantic`:
```suggestion
            workspace_folder = lsp_types.WorkspaceFolder(**workspaceFolders[0]["uri"])
```

The one using `rootUri` already is the best we can do I think.